spec/integration/outputs/kafka_spec.rb

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+
+require "logstash/devutils/rspec/spec_helper"
+require 'logstash/outputs/kafka'
+require 'longshoreman'
+require 'jruby-kafka'
+require 'json'
+require 'poseidon'
+
+CONTAINER_NAME = 'kafka'
+IMAGE_NAME = 'spotify/kafka'
+IMAGE_TAG = 'latest'
+KAFKA_PORT = 9092
+
+RSpec.configure do |config|
+  config.before(:all, :integration => true) do
+    @ls = begin
+            ls = Longshoreman.new
+            ls.container.get(CONTAINER_NAME)
+            ls
+          rescue Docker::Error::NotFoundError
+            Longshoreman.pull_image(IMAGE_NAME, IMAGE_TAG)
+            Longshoreman.new("#{IMAGE_NAME}:#{IMAGE_TAG}", CONTAINER_NAME,
+                             { "ENV" => [ "ADVERTISED_HOST=#{Longshoreman.new.get_host_ip}", "ADVERTISED_PORT=#{KAFKA_PORT}"],
+                               "PortBindings" => { "#{KAFKA_PORT}/tcp" => [{ "HostPort" => "#{KAFKA_PORT}" }]}})
+          end
+    @kafka_host = @ls.get_host_ip
+    @kafka_port = @ls.container.rport(9092)
+  end
+
+  config.after(:suite) do
+    begin
+      ls = Longshoreman::new
+      ls.container.get(CONTAINER_NAME)
+      ls.cleanup
+    rescue Docker::Error::NotFoundError, Excon::Errors::SocketError
+    end
+  end
+end
+
+describe "outputs/kafka", :integration => true do
+  let(:test_topic) { 'test' }
+  let(:config) { {'client_id' => 'spectest', 'bootstrap_servers' => "#{@kafka_host}:#{@kafka_port}", 'topic_id' => test_topic} }
+  let(:event) { LogStash::Event.new({'message' => 'hello', 'topic_name' => 'my_topic', '@timestamp' => LogStash::Timestamp.now}) }
+
+  context 'when outputting messages' do
+    let(:num_events) { 3 }
+
+    it 'should send logstash event to kafka broker' do
+      kafka = LogStash::Outputs::Kafka.new(config)
+      kafka.register
+      num_events.times do kafka.receive(event) end
+      kafka.teardown
+
+      consumer = Poseidon::PartitionConsumer.new("my_test_consumer", @kafka_host, @kafka_port, test_topic, 0, :earliest_offset)
+      messages = consumer.fetch
+      expect(messages.size).to eq(num_events)
```

Comment:
It'd be nice to make the messages the subject and have two specs, one for message count, one for message value.